### PR TITLE
[antithesis] Add test setup for xsvm

### PR DESCRIPTION
**Pull request by marun**
Add an antithesis test setup targeting xsvm to provide an example of subnet use with antithesis and increase coverage for consensus.

## TODO

 - [x] Merge parent #2976
 - [x] Figure out why the docker compose network fails to restart

### CODE REVIEW

File: vms/example/xsvm/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# The version is supplied as a build argument rather than hard-coded
```
[the remark this answers is not in the document]
Comment (StephenButtolph) [reply vms/example/xsvm/Dockerfile:1]: Makes sense to me.

File: tests/antithesis/compose.go [hunk 2]
```diff
@@ -188,3 +228,14 @@ func keyMapToEnvVarMap(keyMap types.Mapping) types.Mapping {
...
+
+// Retrieve the service name for a node at the given index. Common to
+// GenerateComposeConfig and InitDBVolumes to ensure consistency
+// between db volumes configuration and volume paths.
+func getServiceName(index int) string {
+	baseName := "avalanche"
+	if index == 0 {
+		return baseName + "-bootstrap-node"
+	}
+	return fmt.Sprintf("%s-node-%d", baseName, index+1)
```
Comment (StephenButtolph) [inline tests/antithesis/compose.go:240]: Why do we do `index+1`? Seems odd that we would have `avalanche-node-2` but not `avalanche-node-1`

File: tests/antithesis/node_health.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+			log.Printf("node reported healthy at %s", uri)
+			return nil
+		default:
+			log.Printf("node reported unhealthy at %s", uri)
+		}
+
+		select {
+		case <-ticker.C:
+		case <-ctx.Done():
+			return fmt.Errorf("node health check cancelled at %s", uri)
```
Comment (StephenButtolph) [inline tests/antithesis/node_health.go:47]: We should probably wrap `ctx.Err()` here

**Review by StephenButtolph — APPROVED** [review]
lgtm

**Review by abi87 — APPROVED** [review]

File: tests/fixture/subnet/xsvm.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+	}
+
+	if key == nil {
+		var err error
+		key, err = secp256k1.NewPrivateKey()
+		if err != nil {
+			panic(err)
+		}
+	}
+
```
Comment (joshua-kim) [inline tests/fixture/subnet/xsvm.go:28]: I feel like it looks weird to pass in `nil` in the caller function and override it with a generated key. Why don't we just force the caller to generate the key instead?

**Review by joshua-kim — APPROVED** [review]

**Pull request merged**